### PR TITLE
Make the PlatformView client rendering API agnostic.

### PR DESCRIPTION
**Pull request by chinmaygarde**
There is nothing OpenGL specific in the platform view layer now. This allows us to plug in multiple backends per platform.

This is work in progress. Only the Mac platform is done for now.

### CODE REVIEW

**Comment by chinmaygarde** [comment]
I have patched the Android platform. Have also cleaned up the resource context creation.

**Comment by chinmaygarde** [comment]
The Android only updates are in https://github.com/flutter/engine/pull/3099/commits/83606e91189a6c0451f4ec83244e41ffc33e4d94 for reviewability.

File: .clang-format [hunk 1]
```diff
@@ -6,3 +6,4 @@ BasedOnStyle: Chromium
 # 'vector<vector<int>>'. ('Auto' means that clang-format will only use
 # 'int>>' if the file already contains at least one such instance.)
 Standard: Cpp11
+SortIncludes: true
```
Comment (abarth) [inline .clang-format]: Woah!

File: shell/platform/android/platform_view_android.cc [hunk 2]
```diff
@@ -441,46 +107,28 @@ void PlatformViewAndroid::DispatchPointerDataPacket(JNIEnv* env,
...
-
-bool PlatformViewAndroid::SwapBuffers() {
-  TRACE_EVENT0("flutter", "PlatformViewAndroid::SwapBuffers");
-  return context_ != nullptr ? context_->SwapBuffers() : false;
+  return canvas_gl_ ? canvas_gl_->GLOffscreenContextMakeCurrent() : false;
 }
 
 SkISize PlatformViewAndroid::GetSize() {
-  return context_->GetSize();
+  return canvas_gl_ ? canvas_gl_->OnScreenSurfaceSize() : SkISize::Make(0, 0);
```
Comment (abarth) [inline shell/platform/android/platform_view_android.cc]: Skia usually has a `MakeEmpty` or `MakeZero`

File: shell/common/canvas.h [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+ private:
+  bool submitted_;
+
+  virtual bool PerformSubmit() = 0;
+
+  FTL_DISALLOW_COPY_AND_ASSIGN(CanvasFrame);
+};
+
+class Canvas {
```
Comment (chinmaygarde) [inline shell/common/canvas.h]: How about `Surface`?

File: shell/platform/android/platform_view_android.cc (hunk 2 above)
Comment (chinmaygarde) [inline shell/platform/android/platform_view_android.cc]: Done.

Comment (chinmaygarde) [inline shell/platform/android/platform_view_android.cc]: Hmm, apparently not for SkTSize.

**Comment by chinmaygarde** [comment]
All platforms have been updated. @abarth do you want me to change the name of `Canvas` in this patch or another one?

**Comment by abarth** [comment]
If it's not too much work, I'd prefer you change it in this patch given that it's a new class.

**Comment by chinmaygarde** [comment]
@abarth Done and tested all platform. Another look please.

**Pull request merged**